app/services/preserved_object_handler.rb

```diff
@@ -60,7 +60,66 @@ def create
   end
 
   def check_existence
```

Comment:
it's probably work adding a comment such as
`# this is a long, complex method (shameless green);  if it is refactored, revisit the exceptions in rubocop.yml`